[BUG] Sleep runner is failed to complete due to missing client_request_start

**Describe the bug**
BenchmarkError is being raised for sleep operation .

**To Reproduce**
Add following item to schedule, and execute procedure.

```
       {
          "operation": {
            "name": "wait-in-seconds-after-index", 
            "operation-type": "sleep",
            "duration": 2
          }
        },
 ```

**Expected behavior**
Sleep operation should succeed.

**Logs**
If applicable, add logs to help explain your problem.

```
Running delete-target-index                                                    [100% done]
Running create-target-index                                                    [100% done]
Running wait-in-seconds-after-create-index                                     [  0% done]
[ERROR] Cannot execute-test. Error in load generator [0]
        Cannot run task [wait-in-seconds-after-create-index]: 'client_request_start'

```

Log
```

  File "/Users/balasvij/PycharmProjects/knn/opensearch-benchmark/osbenchmark/worker_coordinator/worker_coordinator.py", line 1592, in __call__
    request_start = request_context.request_start

  File "/Users/balasvij/PycharmProjects/knn/opensearch-benchmark/osbenchmark/client.py", line 54, in request_start
    return self.ctx["request_start"]

KeyError: 'request_start'


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/balasvij/PycharmProjects/knn/opensearch-benchmark/osbenchmark/worker_coordinator/worker_coordinator.py", line 1595, in __call__
    client_request_end = request_context.client_request_end

  File "/Users/balasvij/PycharmProjects/knn/opensearch-benchmark/osbenchmark/client.py", line 70, in __aexit__
    client_request_start = self.client_request_start

  File "/Users/balasvij/PycharmProjects/knn/opensearch-benchmark/osbenchmark/client.py", line 62, in client_request_start
    return self.ctx["client_request_start"]

KeyError: 'client_request_start'

```

**More Context (please complete the following information):**
Any workload on OpenSearch 2.x/1.x

**Additional context**
Link to source code  https://github.com/opensearch-project/opensearch-benchmark/blob/main/osbenchmark/worker_coordinator/runner.py#L1666

--- a/osbenchmark/worker_coordinator/runner.py
+++ b/osbenchmark/worker_coordinator/runner.py
@@ -1669,10 +1669,13 @@ class Sleep(Runner):
     """
 
     async def __call__(self, opensearch, params):
+        sleep_duration = mandatory(params, "duration", "sleep")
         opensearch.on_request_start()
         try:
-            await asyncio.sleep(mandatory(params, "duration", "sleep"))
+            request_context_holder.on_client_request_start()
+            await asyncio.sleep(sleep_duration)
         finally:
+            request_context_holder.on_client_request_end()
             opensearch.on_request_end()
 
     def __repr__(self, *args, **kwargs):

--- a/tests/worker_coordinator/runner_test.py
+++ b/tests/worker_coordinator/runner_test.py
@@ -3927,10 +3927,12 @@ class RawRequestRunnerTests(TestCase):
 
 class SleepTests(TestCase):
     @mock.patch("opensearchpy.OpenSearch")
+    @mock.patch('osbenchmark.client.RequestContextHolder.on_client_request_end')
+    @mock.patch('osbenchmark.client.RequestContextHolder.on_client_request_start')
     # To avoid real sleeps in unit tests
     @mock.patch("asyncio.sleep", return_value=as_future())
     @run_async
-    async def test_missing_parameter(self, sleep, opensearch):
+    async def test_missing_parameter(self, sleep, on_client_request_start, on_client_request_end, opensearch):
         r = runner.Sleep()
         with self.assertRaisesRegex(exceptions.DataError,
                                     "Parameter source for operation 'sleep' did not provide the mandatory parameter "
@@ -3938,21 +3940,27 @@ class SleepTests(TestCase):
             await r(opensearch, params={})
 
         self.assertEqual(0, opensearch.call_count)
-        self.assertEqual(1, opensearch.on_request_start.call_count)
-        self.assertEqual(1, opensearch.on_request_end.call_count)
+        self.assertEqual(0, opensearch.on_request_start.call_count)
+        self.assertEqual(0, opensearch.on_request_end.call_count)
+        self.assertEqual(0, on_client_request_start.call_count)
+        self.assertEqual(0, on_client_request_end.call_count)
         self.assertEqual(0, sleep.call_count)
 
     @mock.patch("opensearchpy.OpenSearch")
+    @mock.patch('osbenchmark.client.RequestContextHolder.on_client_request_end')
+    @mock.patch('osbenchmark.client.RequestContextHolder.on_client_request_start')
     # To avoid real sleeps in unit tests
     @mock.patch("asyncio.sleep", return_value=as_future())
     @run_async
-    async def test_sleep(self, sleep, opensearch):
+    async def test_sleep(self, sleep, on_client_request_start, on_client_request_end, opensearch):
         r = runner.Sleep()
         await r(opensearch, params={"duration": 4.3})
 
         self.assertEqual(0, opensearch.call_count)
         self.assertEqual(1, opensearch.on_request_start.call_count)
         self.assertEqual(1, opensearch.on_request_end.call_count)
+        self.assertEqual(1, on_client_request_start.call_count)
+        self.assertEqual(1, on_client_request_end.call_count)
         sleep.assert_called_once_with(4.3)
 
 

